typescript/infra/helm/warp-routes/templates/env-var-external-secret.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: warp-routes-env-var-external-secret
```

Comment:
I think we should probably have the name of this ExternalSecret and the secret name (line 14) be dynamic based on the name of the helm chart / the pod. Otherwise we risk having the many different warp route helm releases try to control the same resources, and bad things can happen. E.g. even if helm doesn't complain about this when doing a deploy, if we end up uninstalling a warp route it'd get rid of this external secret and then break all the other ones that depend on it

I notice that we do this in the key funder helm chart where you probably got this from though - we kinda get away with it there bc key funder is intended to be a singleton (but tbh it's probably a bad practice for us to have hardcoded the name there anyways)

so i'd make this name something like
```
{{ include "hyperlane.fullname" . }}-external-secret
```

and the secret name `{{ include "hyperlane.fullname" . }}-secret` or something